### PR TITLE
v0.2.0 + remove all traces of Gradle Software Model

### PR DESCRIPTION
**Pull request by loosebazooka**
It appears that the Gradle Software Model for java projects is
not moving forward. Change code back to classic plugin format -
using extensions and project callbacks.

This code change affects how to configure the plugin now,
the "appengine" configuration block is now under project, not model.

OLD :
model {
  appengine {
  }
}

NEW :
appengine {
}

### CODE REVIEW

**Comment by loosebazooka** [comment]
@patflynn @meltsufin plugin rewrite.

File: build.gradle [hunk 1]
```diff
@@ -18,7 +18,7 @@ sourceCompatibility = JavaVersion.VERSION_1_7
 targetCompatibility = JavaVersion.VERSION_1_7
 
 group = 'com.google.cloud.tools'
-version = '0.1.2-SNAPSHOT'
+version = '0.2.0-SNAPSHOT'
```
Comment (patflynn) [inline build.gradle]: @meltsufin @loosebazooka 

Why are we still on 0.2.0 if these are 1.0 RC ready?

Comment (meltsufin) [inline build.gradle]: For Maven, I was thinking next release would be 1.0.0.

Comment (loosebazooka) [inline build.gradle]: mostly just because I change a lot of the internals and would like to play around with it for a week before switching to 1.0.0

Comment (loosebazooka) [inline build.gradle]: or does 1.0.0-rc basically imply the same thing?

**Comment by meltsufin** [comment]
install comes up with this warning:
```
warning: [options] bootstrap class path not set in conjunction with -source 1.7
1 warning
```

**Comment by loosebazooka** [comment]
Yeah cause you're probably building with java8. You can build with java7 or bypass that warning by specifying the bootstrap class in your build.gradle

```
compileJavaTask.options.bootClasspath = (new File(java7Home, "/jre/lib/rt.jar")).canonicalPath
```

**Comment by meltsufin** [comment]
LGTM

**Pull request merged**